src/utils/https_proxy_utils/src/https_proxy_utils.c

```diff
@@ -9,12 +9,84 @@
 #include "aduc/logging.h"
 #include "aduc/string_c_utils.h"
 #include <azure_c_shared_utility/crt_abstractions.h> // mallocAndStrcpy_s
-#include <curl/curl.h> // curl_* functions
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
 #include <strings.h>
 
+/**
+ * @brief Implementation of the non-standard strdup
+ * @param proxyOptions str string to duplicate
+ * @return char* duplicated string, deallocate using free. NULL on failure.
+ */
+static char* strdupe(const char* str)
+{
+    char* dupe = malloc(strlen(str) + 1);
```

Comment:
strlen will crash if str is NULL. From usage here, it looks like that'll never happen, but still...